lib/contentful-util.js

```diff
@@ -97,8 +97,25 @@ function resolveLinksInEntry(entry, assets, entries) {
     };
 }
 
+async function syncWithRetry(client, parameters, maxRetries = 5, currentRetry = 1) {
+    try {
+        const result = await client.sync(parameters);
+
+        return result;
+    } catch (error) {
+        if (error.message === 'Request failed with status code 401' && currentRetry < maxRetries) {
```

Comment:
It is usually better to rely on something like error codes rather than arbitrary error message arbitrary error message. Contentful documentation has a page describing all possible error codes that come back with every erroneous response:
https://www.contentful.com/developers/docs/references/errors/

Although it isn't immediately obvious how to acquire the `sys.id` of the error from the caught promise. Doing quick introspection reveals it:
```
error.response.data.sys.id === 'AccessTokenInvalid'
```

Reply:
> It is usually better to rely on something like error codes rather than arbitrary error message arbitrary error message

Agree 100%. I couldn't replicate the issue locally, so I had to do this a bit blindly. I tried to follow the trail of the error in the Contentful package, and I got the impression that the message was the only properly we could rely on. But I may have missed something, so if you're confident that we can extract the error code that way, I'm happy to amend the PR.